models.go

```diff
@@ -50,13 +50,13 @@ type ChatCompletionRequest struct {
 	Messages []ChatCompletionRequestMessage `json:"messages"`
 
 	// What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic
-	Temperature float32 `json:"temperature,omitempty"`
+	Temperature *float32 `json:"temperature,omitempty"`
 
 	// An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered.
-	TopP float32 `json:"top_p,omitempty"`
+	TopP *float32 `json:"top_p,omitempty"`
 
 	// Number of responses to generate
-	N int `json:"n,omitempty"`
+	N *int `json:"n,omitempty"`
```

Comment:
a value of 0 for N is actually invalid and will result in an error from openai, so this one we can potentially leave alone. that way it also doesn't break any existing users of this library.

Reply:
Ah, makes sense! Perhaps when backwards incompatible changes can be introduced (maybe when the big `Engine` function deprecation happens?) the same parameter on `CompletionRequest` should be updated to be a raw number?